pallet-lo-authority-list/src/lib.rs

```diff
@@ -184,17 +206,45 @@ pub mod pallet {
 		pub fn update_legal_officer(
 			origin: OriginFor<T>,
 			legal_officer_id: T::AccountId,
-			data: LegalOfficerData,
+			data: LegalOfficerData<T::AccountId>,
 		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed_or_root(origin.clone())?;
-			if who.is_some() && who.unwrap() != legal_officer_id {
+			if who.is_some() && who.clone().unwrap() != legal_officer_id {
```

Comment:
Just for my understanding, what is the purpose of this added `clone()` ?

Reply:
This prevents the borrowing of `who` so that I can still access the variable later in the code.